flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/BlocklistDeclarativeSlotPool.java

```diff
@@ -90,11 +90,29 @@ public Collection<SlotOffer> registerSlots(
         if (!isBlockedTaskManager(taskManagerLocation.getResourceID())) {
             return super.registerSlots(slots, taskManagerLocation, taskManagerGateway, currentTime);
         } else {
-            LOG.debug("Reject slots {} from a blocked TaskManager {}.", slots, taskManagerLocation);
-            return Collections.emptySet();
+            LOG.debug(
```

Comment:
Maybe move the log to `internalOfferSlotsFromBlockedTaskManager` and also log about the accepted/reject slots?

e.g. Received 3 slots from a blocked TaskManager XXX, 1 was accepted before: [A], 2 was rejected: [B,C].

Reply:
Agree